src/strings/nfa-noodlification.cc

```diff
@@ -234,9 +234,10 @@ SegNfa::NoodleSubstSequence SegNfa::noodlify_mult_eps(const SegNfa& aut, const s
         }
 
         for(const Trans& tr : epsilon_depths_map.at(item.seg_id).at(item.fin)) {
-            Mata::Util::NumberPredicate<Mata::Nfa::State> fins = segments[item.seg_id+1].final; // final states of the segment
+            //TODO: is the use of SparseSet here good? It may take a lot of space. Do you need constant test? Otherwise what about StateSet?
+            Mata::Util::SparseSet<Mata::Nfa::State> fins = segments[item.seg_id+1].final; // final states of the segment
```

Comment:
I gues you could just iterate trhough the original sparse set using its begin and end here ... ?
Maybe just make fins a reference to segments[item.seg_id+1].final, instead of copy?